conan/internal/api/export.py

```diff
@@ -31,18 +31,35 @@ def cmd_export(app, hook_manager, global_conf, conanfile_path, name, version, us
     conanfile.output.scope = conanfile.display_name
     scoped_output = conanfile.output
     # Even though the package_id_non_embed_mode is minor_mode by default,
+    # and package_id_unknown_mode is semver_mode by default,
     # recipes with buggy versions that do not define the attribute will have
     # the same problem regardless
     if not isinstance(ref.version.major.value, int) and ref.version.minor is not None:
+        for mode in ("package_id_non_embed_mode", "package_id_unknown_mode"):
+            if not hasattr(conanfile, mode):
```

Comment:
This seems a bit too repeated with the below? Could we use the inexistence of ``mode`` or the definition with a problematic output to create the same warning message?